core/src/test/scala/unit/kafka/admin/DescribeConsumerGroupTest.scala

```diff
@@ -150,4 +163,228 @@ class DescribeConsumerGroupTest extends KafkaServerTestHarness {
     consumer1Mock.stop()
     consumer2Mock.stop()
   }
+
+  @Test
+  def testDescribeNonExistingGroupWithNewConsumer() {
+    // run one consumer in the group consuming from a single-partition topic
+    val executor = new ConsumerGroupExecutor(brokerList, 1, group, topic)
+
+    // note the group to be queried is a different (non-existing) group
+    val cgcArgs = Array("--bootstrap-server", brokerList, "--describe", "--group", "missing.group")
+    val opts = new ConsumerGroupCommandOptions(cgcArgs)
+    val consumerGroupCommand = new KafkaConsumerGroupService(opts)
+
+    TestUtils.waitUntilTrue(() => {
+        try {
+          val (state, assignments) = consumerGroupCommand.describeGroup()
+          state == Some("Dead") &&
+          assignments == Some(List())
+        } catch {
+          case _: GroupCoordinatorNotAvailableException | _: IllegalArgumentException =>
+            // Do nothing while the group initializes
+            false
+          case e: Throwable =>
+            throw e
+            true
+        }
+      }, "Expected the state to be 'Dead' with no members in the group.")
+
+    consumerGroupCommand.close()
```

Comment:
Well, this actually breaks the unit test, because we don't want to run the `try` block only once. We want to keep trying until the group stabilizes. If we move the `close` to `finally` we close the command after the first try and run into an error on the next try.